scripts/ios-configure-glog.sh

```diff
@@ -20,6 +20,27 @@ if [ -z "$CURRENT_ARCH" ] || [ "$CURRENT_ARCH" == "undefined_arch" ]; then
     fi
 fi
 
+if [ "$CURRENT_ARCH" == "arm64" ]; then
+    cat <<\EOF >>fix_glog_0.3.5_apple_silicon.patch
+diff --git a/config.sub b/config.sub
+index 1761d8b..43fa2e8 100755
+--- a/config.sub
++++ b/config.sub
+@@ -1096,6 +1096,9 @@ case $basic_machine in
+ 		basic_machine=z8k-unknown
+ 		os=-sim
+ 		;;
++	arm64-*)
++		basic_machine=$(echo $basic_machine | sed 's/arm64/aarch64/')
++		;;
+ 	none)
+ 		basic_machine=none-none
+ 		os=-none
+EOF
```

Comment:
Can you move this to a separate file? That can make it easier to read and iterate on in the future.

Reply:
sure! any preference on where that file lives? and do you know where this script is run relative to?

Reply:
Hmm… maybe you can create that file as a sibling to this one (`scripts/`) and reference it by fetching the path to the current file?

Something like this…

```
THIS_DIR=$(cd -P "$(dirname "$(readlink "${BASH_SOURCE[0]}" || echo "${BASH_SOURCE[0]}")")" && pwd)

patch … "$THIS_DIR/fix_glog_0.3.5_apple_silicon.patch"
```

Reply:
@yungsters so i did some experimenting, and the issue is that we aren't actually calling this script directly, but rather reading the contents and passing that directly into the podspec:

https://github.com/facebook/react-native/blob/main/third-party-podspecs/glog.podspec#L14

therefore there is no `${BASH_SOURCE[0]}`, as the script is called directly via `bash -c`, basically eval'ing the contents of the file. so i see three solutions:

1. interpolate the scripts dir path into the script programmatically after we `File.read` the script
2. change the glog.podspec file to call the script directly (i.e. `/bin/bash /absolute/path/to/scripts/ios-configure-glog.sh`), and thus be able to infer that path
3. leave it as is in the PR, dynamically generating the file when needed

do you have a preference? my gut says option 2 because it feels like the "most correct" use of this bash script, but it also is a more fundamental change to how this script is being used and may have ramifications i don't see right now

Reply:
Reading the contents of the file into the podspec? Oh boy… 😬

Thanks for investigating, haha. Hmm… #1 increases complexity so I prefer not doing that.

I agree that #2 seems more right, but I would (maybe incorrectly) presume that this was done a certain way for some reason. But I am all for trying it out!

How about this… I'll merge this PR as-is, and you can optionally follow-up with #2 to improve general engineering of this. If that ends up being a rabbit hole or if you don't have time for it, then we still resolved the issue you set out to initially resolve.